### PR TITLE
Track explicitly when we are initializing config

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java [hunk 1]
```diff
@@ -155,11 +155,18 @@ protected void subscribeAndHandleErrors(ConfigSubscription<?> sub, ConfigKey<?>
      *
      * If the call times out (timeout 1000 ms), no handle will have the changed flag set. You should not configure anything then.
      *
+     * @param isInitializing true if this the config is needed to create the initial configuration for the caller,
```
Comment (jonmv) [inline config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java:158]:
```suggestion
     * @param isInitializing true if this config is needed to create the initial configuration for the caller,
```

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java [hunk 2]
```diff
@@ -227,11 +248,18 @@ public boolean nextGeneration() {
      * If the call times out (timeout 1000 ms), no handle will have the changed flag set. You should not configure anything then.
      *
      * @param timeoutMillis timeout in milliseconds
+     * @param isInitializing true if this the next generation is needed to create the initial configuration for the caller,
```
Comment (jonmv) [inline config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java:251]:
```suggestion
     * @param isInitializing true if this next generation is needed to create the initial configuration for the caller,
```

**Review by jonmv — APPROVED** [review]
Great!

**Review by hmusum — COMMENTED** [review]
👍

**Pull request merged**